azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -350,20 +351,23 @@ private synchronized Project createNewProject(Connection connection,
   }
 
   @Override
-  public void uploadProjectFile(Project project, int version, String filetype,
-      String filename, File localFile, String uploader)
+  public void uploadProjectFile(int projectId, int version, File localFile, String uploader)
```

Comment:
Just wanted to add that the refactor in the JdbcProjectLoader code simply splits the upload method into multiple smaller methods.